src/viam/app/app_client.py

```diff
@@ -717,38 +981,233 @@ async def get_fragment(self, fragment_id: str) -> Fragment:
         response: GetFragmentResponse = await self._app_client.GetFragment(request, metadata=self._metadata)
         return Fragment.from_proto(fragment=response.fragment)
 
-    async def create_fragment(self):
-        raise NotImplementedError()
+    # TODO: Fragments are currently able to have the empty string "" as a name.
+    async def create_fragment(self, name: str, config: Optional[Mapping[str, Any]] = None) -> Fragment:
+        """Create a new private fragment.
 
-    async def update_fragment(self):
-        raise NotImplementedError()
+        Args:
+            name (str): Name of the fragment.
+            config (Optional[Mapping[str, Any]]): _description_
 
-    async def delete_fragments(self):
-        raise NotImplementedError()
+        Raises:
+            GRPCError: If an invalid name is passed.
 
-    async def add_role(self):
-        raise NotImplementedError()
+        Returns:
+            viam.app.app_client.Fragment: The newly created fragment.
+        """
+        organization_id = await self._get_organization_id()
+        request = CreateFragmentRequest(name=name, config=dict_to_struct(config) if config else None, organization_id=organization_id)
+        response: CreateFragmentResponse = await self._app_client.CreateFragment(request, metadata=self._metadata)
+        return Fragment.from_proto(response.fragment)
 
-    async def remove_role(self):
-        raise NotImplementedError()
+    # TODO: The name of a fragment HAS to be updated if any updates to its config and/or public status are being made.
```

Comment:
what is the "TODO" here? can we remove or update this?

Reply:
I thought it was kind of weird that if I want to update a fragment, I have to also change its name (or pass its current name to keep it the same). This is an upstream issue but I feel like it should be made optional